packages/salesforcedx-vscode-core/src/commands/functions/forceFunctionStart/ForceFunctionContainerStartExecutor.ts

```diff
@@ -0,0 +1,139 @@
+import * as vscode from 'vscode';
+import { channelService } from '../../../channels';
+import { notificationService } from '../../../notifications';
+import { FunctionService } from '../functionService';
+import { FUNCTION_RUNTIME_DETECTION_PATTERN } from '../types/constants';
+
+import { getFunctionsBinary } from '@heroku/functions-core';
+import { nls } from '../../../messages';
+import { telemetryService } from '../../../telemetry';
+import { ForceFunctionStartExecutor as ForceFunctionStartExecutor } from './ForceFunctionStartExecutor';
+
+/**
+ * Error types when running SFDX: Start Function
+ * This is also used as the telemetry log name.
+ */
+type ForceFunctionStartErrorType = 'force_function_start_docker_plugin_not_installed_or_started';
+
+const forceFunctionStartErrorInfo: {
```

Comment:
Reading through the code, I see what forceFunctionStartErrorInfo is doing, and looks like this code & its approach originally came from forceFunctionStart.ts.  But reading through the code, my first thought was... "what?"

This (the original code that was carried forward) seems (IMHO) to be a little over engineered.  This might be a good opportunity to clean it up and make it more simple. We could get rid of ForceFunctionStartErrorType and forceFunctionStartErrorInfo, and just change it to:
```
const forceFunctionStartErrorInfo = {
  force_function_start_docker_plugin_not_installed_or_started: {
    cliMessage: 'Cannot connect to the Docker daemon',
    errorNotificationMessage: nls.localize(
      'force_function_start_warning_docker_not_installed_or_not_started'
    )
  }
}
```

...and then further down, not have to do ` (Object.keys(forceFunctionStartErrorInfo) as ForceFunctionStartErrorType[]).forEach(...`

Thoughts?

Reply:
I felt that "what?" I've had to review this a couple of times to understand it and even then it def seem over engineered. I'll simplify. 👍